File: DevExpress VCL/ExpressSpreadSheet/Demos/Delphi/CustomFunctionDemo/CustomFunctionFrm.pas
unit CustomFunctionFrm;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  cxControls, cxSSheet;

type
  TCustomFunctionForm = class(TForm)
    cxSpreadSheet: TcxSpreadSheet;
    procedure FormCreate(Sender: TObject);
  private
  public
    { Public declarations }
  end;

var
  CustomFunctionForm: TCustomFunctionForm;

implementation

{$R *.DFM}


procedure TCustomFunctionForm.FormCreate(Sender: TObject);
var
  CellObject: TcxSSCellObject;
begin
  try
    cxSpreadSheet.LoadFromFile(ExtractFilePath(Application.ExeName) + '\CustomFunctionEx.xls');
    CellObject := cxSpreadSheet.Sheet.GetCellObject(1,4);
    try
      CellObject.Text := '=Area(B3:B4)';
    finally
      CellObject.Free;
    end;
  except
    on E: Exception do
      Application.ShowException(E);
  end;
end;

end.

File: DevExpress VCL/ExpressSpreadSheet/Demos/CBuilder/CustomFunctionDemo/CustomFunctionFrm.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "CustomFunctionFrm.h"
#include "AreaCustomFunc.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxSSheet"
#pragma resource "*.dfm"
TCustomFunctionForm *CustomFunctionForm;
//---------------------------------------------------------------------------
__fastcall TCustomFunctionForm::TCustomFunctionForm(TComponent* Owner)
  : TForm(Owner)
{
  TcxSSCellObject *CellObject;
  try {

    cxSpreadSheet->LoadFromFile(ExtractFilePath(Application->ExeName) + "\CustomFunctionEx.xls");
    CellObject = cxSpreadSheet->Sheet->GetCellObject(1,4);
    try {
      CellObject->Text = "=Area(B3:B4)";
    }
    __finally {
      delete CellObject;
    }
  }
  catch (Exception &err) {
    Application->ShowException(&err);
    ActiveMDIChild->Close();
  }
}
//---------------------------------------------------------------------------
